crates/rustc_codegen_spirv/src/linker/destructure_composites.rs

```diff
@@ -14,29 +14,31 @@ pub fn destructure_composites(function: &mut Function) {
         .all_inst_iter()
         .filter_map(|inst| match inst.class.opcode {
             Op::CompositeConstruct => Some((inst.result_id.unwrap(), inst.clone())),
-            Op::CompositeInsert if inst.operands.len() == 3 => {
+            Op::CompositeInsert => {
                 Some((inst.result_id.unwrap(), inst.clone()))
             }
             _ => None,
         })
         .collect();
     for inst in function.all_inst_iter_mut() {
-        if inst.class.opcode == Op::CompositeExtract && inst.operands.len() == 2 {
+        if inst.class.opcode == Op::CompositeExtract {
             let mut composite = inst.operands[0].unwrap_id_ref();
-            let index = inst.operands[1].unwrap_literal_int32();
+            let index: Vec<u32> = inst.operands[1..].iter().map(|i| i.unwrap_literal_int32()).collect();
```

Comment:
```suggestion
            let indices = &inst.operands[1..];
```

You only need to do the `.unwrap_literal_int32()` on `indices[0]` in the `break` statement.